en/guide/follow_me.md

```diff
@@ -0,0 +1,75 @@
+
+The code fragment below shows how to set the configuration:
+```cpp
+// configure follow me behaviour
+FollowMe::Config config;
+config.min_height_m = 12.f;  // Minimum height
+config.follow_dist_m = 20.f;  // Follow distance
+config.responsiveness = 0.2f;  // Higher responsiveness
+config.follow_direction = FollowMe::Config::FollowDirection::FRONT;  //Follow from front-centre
+
+// Apply configuration
+bool configured = device.follow_me().set_config(config);
+if (configured) {
+    // handle config failure
+    std::cout << "Configuration failed" << std::endl;
+}
+```
+
+The [get_config()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a054aebafe0839a1028f277285b769fe5) method is provided to get the current configuration:
+```cpp
+auto curr_config = device.follow_me().get_config();
+```
+
+## Following a Target
+
+To start and stop following a target, call [start()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a694749d43d527f85584df25a49b05ccf) and [stop()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1a6394507b0fb96bceebe6efd17f0529ce), respectively.
+
+In between the calls, use [set_curr_target_location()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1afb8c24ed93421e904b0f528569c7699a) to set the target position for the vehicle to follow. 
+DroneCore automatically resends the last set position at the rate required by the autopilot, but this can be overridden by setting the location at another rate.
+
+> **Note** The API does not automatically get the position of the device running the DroneCore. The client app must fetch this using OS-specific methods and pass them to the `FollowMe` class.
+
+```cpp
+// Start following
+FollowMe::Result follow_me_result = device.follow_me().start();
+if (follow_me_result != FollowMe::Result::SUCCESS) {
+    // handle start failure (in this case print error)
+    std::cout << "Failed to start following:" << FollowMe::result_str(follow_me_result) << std::endl;
+}
+
+// ... Get target position from underlying platform or wherever and supply to vehicle 
+// Here we just show one point being set. Minimum values specified in the configuration are respected.
+follow_me.set_curr_target_location({ 47.39776569, 8.54553292, 9.0, 0.f, 0.f, 0.f });
+
+// Stop following
+follow_me_result = device.follow_me().stop();
+if (follow_me_result != FollowMe::Result::SUCCESS) {
+    // handle stop failure (in this case print error)
+    std::cout << "Failed to stop following:" << FollowMe::result_str(follow_me_result) << std::endl;
+}
+```
+
+The last location that was set can be retrieved using [get_last_location()](../api_reference/classdronecore_1_1_follow_me.md#classdronecore_1_1_follow_me_1ab68273d5ace65ee953afa1797ae49e7c).

```

Comment:
@shakthi-prashanth-m What happens if you haven't set any location when you select `start()`? Do you have to set a start location before starting the API.

Reply:
@hamishwillee It is optional to set location before starting FollowMe; in that case, `get_last_location()` would return `NAN` values. In the [second integration test of FollowMe](https://github.com/dronecore/DroneCore/blob/develop/integration_tests/follow_me.cpp#L79-L147), there's no location set at all.

Reply:
Thanks. So what actually happens - the API doesn't send anything, or PX4 just ignores whatever messages it gets?

Reply:
When we start FollowMe => We send `MAV_CMD_DO_SET_MODE` to change to `PX4_CUSTOM_SUB_MODE_AUTO_FOLLOW_TARGET` mode.
If there's no position set, nothing happens at Vehicle side other than being at FollowMe mode.
When we start pushing GPS location updates, they're sent to Vehicle.

Reply:
Thanks!